### ISSUE TITLE
Skip comments for delimiter detection

### ISSUE THREAD

**Issue by candu** [mholt/PapaParse#489]
I have CSV files with more than 10 comment lines at the beginning, which trips up `guessDelimiter()` - since that function uses a `Parser` with `preview: 10`, it never sees the data rows.

`guessDelimiter()` should probably respect the `comments:` config option here.

**Comment by candu** [comment]
As a note to anyone else who runs across this, I used `beforeFirstChunk()` to work around this issue:

```
    // rest of config...
    beforeFirstChunk(chunk) {
        let i = 0;
        while (i < chunk.length && chunk[i] === commentPrefix) {
          while (i < chunk.length && !isLineEndingCharacter(chunk[i])) {
            i++;
          }
          while (i < chunk.length && isLineEndingCharacter(chunk[i])) {
            i++;
          }
        }
        return chunk.slice(i);
      }
```

**Comment by pokoli** [comment]
A pull request to skip comments on guest delimiters will be very welcome.

**Comment by pokoli** [comment]
Fixed in #503
